src/dash-adapter.js

```diff
@@ -403,6 +408,43 @@ export default class DashAdapter extends BaseMediaSourceAdapter {
     }
   }
 
+  /**
+   * attach media - return the media source to handle the video tag
+   * @public
+   * @param {boolean} playbackEnded don't seek to the last detach point
+   * @returns {void}
+   */
+  attachMediaSource(playbackEnded: ?boolean): void {
+    if (!this._shaka) {
+      if (this._videoElement && this._videoElement.src) {
+        Utils.Dom.setAttribute(this._videoElement, 'src', '');
+        Utils.Dom.removeAttribute(this._videoElement, 'src');
+      }
+      this._init();
+      if (!isNaN(this._lastTimeDetach) && !playbackEnded) {
+        const canPlayHandler = () => {
+          this._videoElement.removeEventListener(Html5EventType.CAN_PLAY, canPlayHandler);
+          this.currentTime = this._lastTimeDetach;
+          this._lastTimeDetach = NaN;
+        };
+        this._videoElement.addEventListener(Html5EventType.CAN_PLAY, canPlayHandler);
+      }
+    }
+  }
+  /**
+   * detach media - will remove the media source from handling the video
+   * @public
+   * @returns {void}
+   */
+  detachMediaSource(): void {
+    if (this._shaka) {
+      this._lastTimeDetach = this.currentTime;
+      this._reset().then(() => {
+        this._shaka = null;
```

Comment:
should probably go into `_reset`

Reply:
I'm not sure, I'm using _reset on destroy as well and there it returns promise from destroy